process/process.go

```diff
@@ -171,11 +178,37 @@ func (p *Process) Start() error {
 				}
 			}
 
-			lineCallbackWaitGroup.Add(1)
-			go func(line string) {
-				defer lineCallbackWaitGroup.Done()
-				p.LineCallback(line)
-			}(string(line))
+			// If we're timestamping this main thread will take
+			// the hit of running the regex so we can build up
+			// the timestamped buffer without breaking headers,
+			// otherwise we let the goroutines take the perf hit.
+
+			checkedForCallback := false
+			lineHasCallback := false
```

Comment:
This bit here has me a little confused. `lineHasCallback` always seems to be false unless timestamping has been turned on?